src/main/java/org/gridsuite/dynamicsecurityanalysis/server/service/DynamicSecurityAnalysisWorkerService.java

```diff
@@ -0,0 +1,311 @@
+import static org.gridsuite.dynamicsecurityanalysis.server.utils.Utils.*;
+
+/**
+ * @author Thang PHAM <quyet-thang.pham at rte-france.com>
+ */
+@ComponentScan(basePackageClasses = {NetworkStoreService.class, NotificationService.class})
+@Service
+public class DynamicSecurityAnalysisWorkerService extends AbstractWorkerService<SecurityAnalysisReport, DynamicSecurityAnalysisRunContext, DynamicSecurityAnalysisParametersInfos, DynamicSecurityAnalysisResultService> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DynamicSecurityAnalysisWorkerService.class);
+
+    private final DynamicSimulationClient dynamicSimulationClient;
+    private final ActionsClient actionsClient;
+    private final ParametersService parametersService;
+
+    public DynamicSecurityAnalysisWorkerService(NetworkStoreService networkStoreService,
+                                                NotificationService notificationService,
+                                                ReportService reportService,
+                                                ExecutionService executionService,
+                                                DynamicSecurityAnalysisObserver observer,
+                                                ObjectMapper objectMapper,
+                                                DynamicSecurityAnalysisResultService dynamicSecurityAnalysisResultService,
+                                                DynamicSimulationClient dynamicSimulationClient,
+                                                ActionsClient actionsClient,
+                                                ParametersService parametersService) {
+        super(networkStoreService, notificationService, reportService, dynamicSecurityAnalysisResultService, executionService, observer, objectMapper);
+        this.dynamicSimulationClient = Objects.requireNonNull(dynamicSimulationClient);
+        this.actionsClient = Objects.requireNonNull(actionsClient);
+        this.parametersService = Objects.requireNonNull(parametersService);
+    }
+
+    /**
+     * Use this method to mock with DockerLocalComputationManager in case of integration tests with test container
+     *
+     * @return a computation manager
+     */
+    public ComputationManager getComputationManager() {
+        return executionService.getComputationManager();
+    }
+
+    @Override
+    protected DynamicSecurityAnalysisResultContext fromMessage(Message<String> message) {
+        return DynamicSecurityAnalysisResultContext.fromMessage(message, objectMapper);
+    }
+
+    public void updateResult(UUID resultUuid, SecurityAnalysisReport result) {
+        Objects.requireNonNull(resultUuid);
+        DynamicSecurityAnalysisStatus status = result.getResult().getPostContingencyResults().stream().anyMatch(postContingencyResult -> postContingencyResult.getStatus() == PostContingencyComputationStatus.FAILED) ?
+                DynamicSecurityAnalysisStatus.FAILED :
```

Comment:
You're sure of this rule ? One fail for one contingency => the computation is in failure ?

Reply:
dynamic-security-analysis result is the same structure as normal SA but is interested in PostContingencyResult rather than PreContingencyResult...